src/event_watcher/windows.rs

```diff
@@ -160,8 +188,83 @@ unsafe extern "system" fn wndproc(
     DefWindowProcA(window, message, wparam, lparam)
 }
 
+unsafe extern "system" fn win_event_hook_proc(
+    _h_win_event_hook: HWINEVENTHOOK,
+    _event: u32,
+    hwnd: HWND,
+    _id_object: i32,
+    _id_child: i32,
+    _id_event_thread: u32,
+    _dwms_event_time: u32,
+) {
+    let mut window_rect = RECT::default();
+    if !GetWindowRect(hwnd, &mut window_rect).is_ok() {
+        return;
+    }
+
+    let hmonitor: HMONITOR = MonitorFromWindow(hwnd, MONITOR_DEFAULTTONULL);
+    if hmonitor.is_invalid() {
+        return;
+    }
+
+    let mut info = MONITORINFOEXW::default();
+    info.monitorInfo.cbSize = size_of::<MONITORINFOEXW>() as u32;
+    let info_ptr = &mut info as *mut _ as *mut MONITORINFO;
+
+    if !GetMonitorInfoW(hmonitor, info_ptr).as_bool() {
+        return;
+    }
+
+    let is_fullscreen = window_rect.left == info.monitorInfo.rcMonitor.left
+        && window_rect.top == info.monitorInfo.rcMonitor.top
+        && window_rect.right == info.monitorInfo.rcMonitor.right
+        && window_rect.bottom == info.monitorInfo.rcMonitor.bottom;
+
+    // TODO: EnumDisplayDevicesW to get the actual device paths?
+    let display_name = wchar_to_string(&info.szDevice);
+
+    let window_data_ptr = GLOBAL_WINDOW_DATA.load(Ordering::SeqCst);
+    if window_data_ptr.is_null() {
+        return;
+    }
+
+    let window_data = &mut *window_data_ptr;
+
+    // TODO: This is likely to be buggy on multi monitor setups when both monitors have fullscreen apps open
```

Comment:
this is a tricky case to handle, but having one global bool for fullscreen is wrong, as it's possible to get it stuck in wrong case when switching from fullscreen app on one monitor, to non-fullscreen app on another.